src/audio/kpb.c

```diff
@@ -1130,9 +1135,13 @@ static void kpb_init_draining(struct comp_dev *dev, struct kpb_client *cli)
 		kpb->draining_task_data.dev = dev;
 		kpb->draining_task_data.sync_mode_on = kpb->sync_draining_mode;
 
-		/* Set host-sink copy mode to blocking */
-		comp_set_attribute(kpb->host_sink->sink, COMP_ATTR_COPY_TYPE,
-				   &copy_type);
+		/* save current sink copy type */
+		comp_get_attribute(kpb->host_sink->sink, COMP_ATTR_COPY_TYPE,
+				   &kpb->draininig_task_data.copy_type);
```

Comment:
I feel like there's a typo here?

Reply:
@paulstelian97 yes, spotted by the CI also, I change it now so that it properly works.